### PR TITLE
[Target] Remove deprecated parameters from target

### LINKED ISSUE #12409: [Bug] _reconstruct_from_deprecated_options creates incorrect executor
example:
```
target = tvm.target.Target("llvm -link-params=1")
executor = Executor("aot")
```
This will result in this executor: `graph{"link-params": 1}`

I looked at the logic of `_reconstruct_from_deprecated_options`, the issue is that if the executor config is not part of the target it will assume `graph` by default. This is obviously incorrect.

Possible solutions
- Pass `executor` to `_reconstruct_from_deprecated_options` and derive the final executor considering the original executor that was passed to build.
- Remove this function and instead show runtime error when wrong config is passed.

I suggest the second approach because the first approach is fairly complicated considering the number of configs and most likely it will result in wrong combination. This could be really annoying considering the fact that you could end up building with incorrect executor without noticing.

thoughts?

cc @areusch @Mousius

### PR DESCRIPTION
**Pull request by mehrdadh**
fixes #12409 

cc @areusch @Mousius

### CODE REVIEW

File: tests/python/relay/test_build_module.py [hunk 1]
```diff
@@ -22,48 +22,23 @@
...
-        [
-            Target(
-                "c -executor=aot -link-params=1 -interface-api=c"
-                "  -unpacked-api=1 -runtime=c -system-lib"
-            ),
-            Executor("aot", {"unpacked-api": 1, "interface-api": "c"}),
-            Runtime("crt", {"system-lib": True}),
-        ],
-    ],
+test_target = tvm.testing.parameter(
```
Comment (Mousius) [inline tests/python/relay/test_build_module.py:27]: We should probably keep using `pytest.mark.parametrize` to aid in linting this file.

File: tests/python/relay/test_build_module.py [hunk 2]
```diff
@@ -22,48 +22,23 @@
...
 
-    assert (executor is None and actual_executor is None) or (executor.name == actual_executor.name)
-    # sort as TVM Map cannot guarantee round-trip order.
-    assert (executor is None and actual_executor is None) or (
-        sorted(executor.attrs.items()) == sorted(actual_executor.attrs.items())
-    )
-    assert runtime == actual_runtime
+
+def test_deprecated_target_parameters(test_target, target):
+    unsupported_config = test_target.split(" ")[1].replace("=", " ").split(" ")[0][1:]
```
Comment (Mousius) [inline tests/python/relay/test_build_module.py:38]: This is pretty complex, would be better to have `test_target` and `unsupported_config` in a tuple.

**Review by Mousius — CHANGES_REQUESTED** [review]

File: tests/python/relay/test_build_module.py (hunk 2 above)
Comment (mehrdadh) [reply tests/python/relay/test_build_module.py]: done.

File: apps/wasm-standalone/wasm-graph/tools/build_graph_lib.py [hunk 3]
```diff
@@ -72,7 +72,7 @@ def build_graph_lib(opt_level):
     shape_dict = {input_name: img_data.shape}
 
     mod, params = relay.frontend.from_onnx(onnx_model, shape_dict)
-    target = "llvm -mtriple=wasm32-unknown-unknown -mattr=+simd128 --system-lib"
+    target = "llvm -mtriple=wasm32-unknown-unknown -mattr=+simd128"
```
Comment (areusch) [inline apps/wasm-standalone/wasm-graph/tools/build_graph_lib.py:75]: i think we need to keep --system-lib passed to Executor config here. can you audit the other places you removed `--system-lib` and ensure the Executor passed to relay.build has it?

**Review by areusch — COMMENTED** [review]
thanks @mehrdadh, broadly looks good but we should check on `system-lib`, because that changes the way code is loaded on the target.

**Comment by mehrdadh** [comment]
@Mousius PTAL, thanks!

**Review by Mousius — APPROVED** [review]
LGTM! The failing test seems unrelated, thanks for picking this up @mehrdadh 😸

**Comment by masahi** [comment]
So `--link-params=1` in target is deprecated now? I am about to send a PR that depends on `link-params` option in target...

**Comment by areusch** [comment]
@masahi I see...I think perhaps we might need to arrange to thread the Executor through. at the moment the way we model memory in the compiler doesn't guarantee that Constant will wind up in any particular location, so link-params is mainly an executor-level thing. what do you think? the change that caused this was also a relay.build API change as well.

**Comment by masahi** [comment]
@areusch Since most common use cases do not need `link-params`, I hesitate to break API and impose `executor` argument to all users of meta schedule task extraction. Instead, I'm leaning toward pushing the responsibility to users if they want to use `link-params` together with meta schedule, by requiring the following usage:

```
    with tvm.transform.PassContext(config={"relay.FuseOps.link_params": 1}):
         meta_schedule.extract_task_from_relay(...)
```

**Pull request merged**